### PR TITLE
Arm

### PR DESCRIPTION
**Pull request by JoystickMaster-race**
**What this PR does**
First version of Alpha Arm (wrist will probably be weird)

**What does it fix/add?**
_Reference issues with "Fixes #10" or "Closes #10", where #10 is the number assigned to the issue_

**What does this depend on?**
_References pull requests or branches/forks that this pull request depends on_

### CODE REVIEW

File: src/main/include/AlphaArm.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// Open Source Software, you can modify it according to the terms
+// of the MIT License at the root of this project
+
+#pragma once
+#include <frc/DigitalInput.h>
+
+#include "Wombat.h"
+
+struct AlphaArmConfig {
+  // wom::Gearbox armGearBox;
```
Comment (spacey-sooty) [inline src/main/include/AlphaArm.h:11]: remove

File: src/main/include/AlphaArm.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  wom::Gearbox alphaArmGearbox;
+  wom::Gearbox wristGearbox;
+};
+
+enum class AlphaArmState { kIdle, kIntakeAngle, kAmpAngle, kSpeakerAngle, kRaw };
+
+class AlphaArm : public ::behaviour::HasBehaviour {
+ public:
+  explicit AlphaArm(AlphaArmConfig config);
+
```
Comment (spacey-sooty) [inline src/main/include/AlphaArm.h:21]: do we wanna add an OnStart with zeroing?

File: src/main/include/AlphaArm.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  void OnUpdate(units::second_t dt);
+  void SetRaw(units::volt_t volt);
```
Comment (spacey-sooty) [inline src/main/include/AlphaArm.h:23]: Rename to SetVolts or SetRawVolts
Should probably take in a voltage for each Voltage variable

File: src/main/cpp/AlphaArm.cpp [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      _config.alphaArmGearbox.motorController->SetVoltage(0_V);
+      _config.wristGearbox.motorController->SetVoltage(0_V);
+
+      break;
+    case AlphaArmState::kRaw:
+      setAlphaArmVoltage = _armVoltage;
+      break;
+    default:
+      std::cout << "oops, wrong state" << std::endl;
+      break;
```
Comment (spacey-sooty) [inline src/main/cpp/AlphaArm.cpp:22]: Doesn't handle all states

File: src/main/cpp/AlphaArm.cpp [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+void AlphaArm::OnUpdate(units::second_t dt) {
+  switch (_state) {
+    case AlphaArmState::kIdle:
+      // transmission translate
+      _config.alphaArmGearbox.motorController->SetVoltage(0_V);
+      _config.wristGearbox.motorController->SetVoltage(0_V);
+
+      break;
+    case AlphaArmState::kRaw:
+      setAlphaArmVoltage = _armVoltage;
```
Comment (spacey-sooty) [inline src/main/cpp/AlphaArm.cpp:18]: Why would we set another variable for this?

File: src/main/cpp/AlphaArm.cpp [hunk 6]
```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2023-2024 CurtinFRC
+// Open Source Software, you can modify it according to the terms
+// of the MIT License at the root of this project
+
+#include "AlphaArm.h"
+
+AlphaArm::AlphaArm(AlphaArmConfig config) : _config(config) {}
+
+void AlphaArm::OnUpdate(units::second_t dt) {
+  switch (_state) {
```
Comment (spacey-sooty) [inline src/main/cpp/AlphaArm.cpp:10]: Doesn't handle all states

File: src/main/cpp/AlphaArm.cpp [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  _config.alphaArmGearbox.motorController->SetVoltage(setAlphaArmVoltage);
+  _config.wristGearbox.motorController->SetVoltage(setWristVoltage);
+}
+
+void AlphaArm::SetState(AlphaArmState state) {
+  _state = state;
+}
+
+void AlphaArm::SetRaw(units::volt_t voltR) {
+  _voltR = voltR;
```
Comment (spacey-sooty) [inline src/main/cpp/AlphaArm.cpp:35]: why are these named voltR?

File: src/main/cpp/AlphaArmBehaviour.cpp [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  Controls(alphaArm);
+}
+
+void AlphaArmManualControl::OnTick(units::second_t dt) {
+  if (_codriver->GetXButtonPressed()) {
+    if (_rawControl == true) {
+      _rawControl = false;
+    } else {
+      _rawControl = true;
+    }
```
Comment (spacey-sooty) [inline src/main/cpp/AlphaArmBehaviour.cpp:20]: use _rawControl = !_rawControl instead

File: src/main/cpp/AlphaArmBehaviour.cpp [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
```
Comment (spacey-sooty) [inline src/main/cpp/AlphaArmBehaviour.cpp:22]: Handle other states as well

File: src/main/include/AlphaArmBehaviour.h [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+class AlphaArmManualControl : public behaviour::Behaviour {
+ public:
+  explicit AlphaArmManualControl(AlphaArm* alphaArm, frc::XboxController* codriver);
+  void OnTick(units::second_t dt);
+
+ private:
+  AlphaArm* _alphaArm;
+  frc::XboxController* _codriver;
+  // units::volt_t _rightStick = ((_codriver->GetRightY()>0.05 || _codriver->GetRightY() < -0.05
+  // )?_codriver->GetRightY():0) * 2_V;
```
Comment (spacey-sooty) [inline src/main/include/AlphaArmBehaviour.h:22]: remove

**Review by spacey-sooty — CHANGES_REQUESTED** [review]

File: src/main/cpp/AlphaArm.cpp (hunk 7 above)
Comment (JoystickMaster-race) [reply src/main/cpp/AlphaArm.cpp:35]: oops, mb

**Pull request merged**